### ISSUE TITLE
setup-gcloud failing on windows 2019 github runners

### ISSUE THREAD

**Issue by SC6333** [google-github-actions/setup-gcloud#655]
### TL;DR

Hi,

I am seeing a recent failure with the setup-gcloud step in windows 2019 runner with the following error message

google-github-actions/setup-gcloud failed with: failed to execute command `gcloud.cmd --quiet auth _***gcloud_email****_ --key-file -`: The system cannot find the path specified.

I am running the auth step before it and I can see the key file has been generated and is present in the right path.
The same sequence of steps also works when I switch the runner to an ubuntu one.
I am using python 3.9 in the setup and a regular windows 2019 github runner.

Can you help with this?

### Expected behavior

Gcloud gets successfully setup in the windows runner

### Action YAML

```yaml
name: Builds SDK_C

on:
  push:
    branches:
      - master
      - MPSDK-2185
  pull_request:
    branches:
      - master

env:
  BRANCH_NAME: ${{ github.head_ref || github.ref_name }}

jobs:
  Build-and-Upload-SDK:
    runs-on: windows-2019
    steps:

      - name: Checkout code
        uses: actions/checkout@v3.3.0
        with:
          fetch-depth: 0
      - name: Set Build Number
        run: echo "BUILD_NUMBER=$( git rev-list --count HEAD )" >> $env:GITHUB_ENV

      - name: Setup python
        uses: actions/setup-python@v4
        with:
          python-version: '3.9'

      - name: Export gcloud related env variable
        run: echo "CLOUDSDK_PYTHON=/usr/bin/python" >> $env:GITHUB_ENV

      - name: Authenticate to Google Cloud
        uses: google-github-actions/auth@v1
        with:
          credentials_json: ${{ secrets.GCE_SA_KEY }}
          project_id: spg-zpc-tools
          export_environment_variables: true
      
      - name: Show Dir
        run: dir

      - name: Setup Google Cloud SDK
        uses: google-github-actions/setup-gcloud@v0

      - name: Setup MSBuild
        uses: microsoft/setup-msbuild@v1.3.1

      - name: Setup Android SDK
        uses: android-actions/setup-android@v2
      - name: Setup .NET Core SDK
        uses: actions/setup-dotnet@v3.0.3
        with:
          dotnet-version: '5.0.x'
      - name: Setup NuGet
        uses: NuGet/setup-nuget@v1.1.1

      - name: Setup VSTest.console.exe
        uses: darenm/Setup-VSTest@v1.2

      - name: Setup SHFB
        uses: Bassman2/setup-SHFB@v1
        with:
          version: 2022.10.15.0

      - name: Install Community Tasks
        run: |
          choco install msbuild.communitytasks
```

**Comment by sethvargo** [comment]
Hi @SC6333 - we are in the process of dropping support for `windows-2019` because the version of Python that is installed on the system does not meet the minimum requirements for modern gcloud versions. You can either:

- Pin to a [gcloud version 447.0.0 or earlier](https://cloud.google.com/sdk/docs/release-notes#44700_2023-09-19)

- Use `actions/setup-python@v4` with Python 3.8+ and export the [python-path](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#outputs-and-environment-variables) output value as the `CLOUDSDK_PYTHON` environment variable

**Comment by SC6333** [comment]
Hi @sethvargo ,

I have already used the setup-python step to install python 3.9 and also set the CLOUDSDK_PYTHON path as shown in my Action YAML file.
After your comment I tried using the gcloud version to 447.0.0 as well. But the error remains same. This error is occurring in the windows machine. The issue was replicated on a windows-latest machine as well.
Can you help shed some light on why its happening?

TIA

**Comment by sethvargo** [comment]
Your action.yml is running:

```sh
CLOUDSDK_PYTHON=/usr/bin/python >> $env:GITHUB_ENV
```

but it needs to be something like:

```sh
CLOUDSDK_PYTHON=${steps.setup-python.outputs.python-path} >> $env:GITHUB_ENV
```

You will need to give your `actions/setup-python` step an id of `setup-python` so the output reference will work.

**Comment by SC6333** [comment]
Thank you that seems to have worked, Thank you
